test/test_cases/run_dry_rising_bubble_3d.jl

```diff
@@ -31,31 +35,169 @@ function run_dry_rising_bubble_3d(
         parameters = params,
         hyperdiffusivity = FT(100),
     )
+    model_pottemp = Nonhydrostatic3DModel(
+        domain = domain,
+        boundary_conditions = nothing,
+        thermodynamics = PotentialTemperature(),
+        parameters = params,
+        hyperdiffusivity = FT(100),
+    )
 
     # execute differently depending on testing mode
     if mode == :integration
         # TODO!: run with input callbacks = ...
-        simulation = Simulation(model, stepper, dt = dt, tspan = (0.0, 1.0))
-        @test simulation isa Simulation
+        @testset "Integration: Potential Temperature Model" begin
+            simulation =
+                Simulation(model_pottemp, stepper, dt = dt, tspan = (0.0, 1.0))
+            @test simulation isa Simulation
+            # test set function
+            @unpack ρ, uh, w, ρθ = init_dry_rising_bubble_3d(FT, params, :ρθ)
+            set!(simulation, :base, ρ = ρ, uh = uh, w = w)
+            set!(simulation, :thermodynamics, ρθ = ρθ)
+            # test error handling
+            @test_throws ArgumentError set!(simulation, quack = ρ)
+            @test_throws ArgumentError set!(simulation, ρ = "quack")
+            # test successful integration
+            @test step!(simulation) isa Nothing # either error or integration runs
+        end
+        @testset "Integration: Total Energy Model" begin
+            simulation = Simulation(model, stepper, dt = dt, tspan = (0.0, 1.0))
+            @test simulation isa Simulation
+            # test set function
+            @unpack ρ, uh, w, ρe_tot =
+                init_dry_rising_bubble_3d(FT, params, :ρe_tot)
+            set!(simulation, :base, ρ = ρ, uh = uh, w = w)
+            set!(simulation, :thermodynamics, ρe_tot = ρe_tot)
+            # test error handling
+            @test_throws ArgumentError set!(simulation, quack = ρ)
+            @test_throws ArgumentError set!(simulation, ρ = "quack")
+            # test successful integration
+            @test step!(simulation) isa Nothing # either error or integration runs
+        end
+    elseif mode == :regression
+        @testset begin
+            "Regression: Potential Temperature Model"
+            simulation =
+                Simulation(model_pottemp, stepper, dt = dt, tspan = (0.0, 1.0))
+            @unpack ρ, uh, w, ρθ = init_dry_rising_bubble_3d(FT, params, :ρθ)
+            set!(simulation, :base, ρ = ρ, uh = uh, w = w)
+            set!(simulation, :thermodynamics, ρθ = ρθ)
+            u = simulation.integrator.u
+            ∫ρ_0 = sum(u.base.ρ)
+            ∫ρθ_0 = sum(u.thermodynamics.ρθ)
+            step!(simulation)
+            u = simulation.integrator.u
+            # perform regression check
+            current_min = 299.9999999523305
+            current_max = 300.468563373248
+            θ = u.thermodynamics.ρθ ./ u.base.ρ
 
```

Comment:
Can we have a slightly more informative regression test? As I see it the bubble is evolved for one time step and therefore doesn't change very much. So we are effectively regression testing the initial condition here.